target/linux/realtek/files-6.12/drivers/net/dsa/rtl83xx/rtl930x.c

```diff
@@ -296,6 +296,25 @@ static void rtl930x_l2_learning_setup(void)
 	sw_w32((0x7fff << 2) | 0, RTL930X_L2_LRN_CONSTRT_CTRL);
 }
 
+static void rtl930x_enable_learning(int port, bool enable)
+{
+	/* Limit learning to maximum: 32k entries */
+
+	sw_w32_mask(0x7fff << 3, enable ? (0x7ffe << 3) : 0,
+		    RTL930X_L2_LRN_PORT_CONSTRT_CTRL + (port << 2));
+}
+
+static void rtl930x_enable_flood(int port, bool enable)
+{
+	/* 0: Forward
+	 * 1: Drop
+	 * 2: to CPU
+	 * 3: Copy to CPU
+	 */
+	sw_w32_mask(0x7, enable ? 0 : 1,
+		    RTL930X_L2_LRN_PORT_CONSTRT_CTRL + (port << 2));
+}
+
```

Comment:
Please use GENMASK where possible and use rtldsa prefix. Same goes for RTL931x functions.